### PR TITLE
Allow coverage to be run on localhost

### PR DESCRIPTION
**Pull request by jayudey-wf**
## Issue
- #89 

## Changes
**Source:**
- Start a pub serve process and run browser tests through there for coverage purposes

**Tests:**
- Current tests provide adequate coverage

## Areas of Regression
- Coverage

## Testing
- Run coverage and ensure that browser based tests are still being reported

### CODE REVIEW

**Comment by dustinlessard-wf** [comment]
+1

File: lib/src/tasks/coverage/api.dart [hunk 1]
```diff
@@ -315,6 +331,35 @@ class CoverageTask extends Task {
     coverage.createSync();
     coverage.writeAsStringSync(JSON.encode(mergedJson));
     return coverage;
+
+//    Map mergedJson = {};
```
Comment (trentgrover-wf) [inline lib/src/tasks/coverage/api.dart]: why all the commented code?

File: test/integration/coverage_test.dart [hunk 2]
```diff
@@ -49,23 +60,23 @@ void main() {
...
-    }, timeout: new Timeout(new Duration(seconds: 60)));
-
-    test('should generate coverage for VM tests', () async {
-      expect(await runCoverage(projectWithVmTests), isTrue);
-      File lcov = new File('$projectWithVmTests/coverage/coverage.lcov');
-      expect(lcov.existsSync(), isTrue);
-    }, timeout: new Timeout(new Duration(seconds: 60)));
+    }, timeout: new Timeout(new Duration(seconds: 300)));
 
+//    test('should generate coverage for VM tests', () async {
```
Comment (trentgrover-wf) [inline test/integration/coverage_test.dart]: still expecting to bring these commented out tests back?

**Comment by trentgrover-wf** [comment]
dartfmt fail

**Comment by jayudey-wf** [comment]
troubleshooting effort fails

File: test/integration/coverage_test.dart (hunk 2 above)
Comment (jayudey-wf) [inline test/integration/coverage_test.dart]: yeah, we just need to figure out how to get the browser tests to work on travis

**Comment by greglittlefield-wf** [comment]
@jayudey-wf Whoops, looks like there's been some overlap here...

https://github.com/Workiva/dart_dev/pull/92 Includes functionality to start a pub server when running coverage.

**Comment by jayudey-wf** [comment]
@greglittlefield-wf I'll close mine as yours looks more comprehensive and I can't figure out why mine is failing on travis...

**Comment by greglittlefield-wf** [comment]
Can't figure out why mine is failing, either... :smile:

**Pull request closed without merging**